aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -214,27 +213,74 @@ Tensor rsub(const Tensor& self, Scalar other, Scalar alpha) {
   return native::rsub(self, wrapped_scalar_tensor(other), alpha);
 }
 
-Tensor& logical_xor_out(Tensor& result, const Tensor& self, const Tensor& other) {
+// ~~~~~~~~~~~~~ Binary logical operators BEGIN ~~~~~~~~~~~~~~~~~
+
+DEFINE_DISPATCH(logical_and_stub);
+DEFINE_DISPATCH(logical_or_stub);
+DEFINE_DISPATCH(logical_xor_stub);
+
+template <typename Stub>
+static inline Tensor& logical_binary_out_impl(Tensor& result, const Tensor& self, const Tensor& other, Stub& stub) {
```

Comment:
In my latest change, I changed `Stub` to `Stub&`, as I've noticed that a `Stub` object caches some results in its `()` operator and I think we should use an lvalue reference here, which shouldn't hurt even if it doesn't help. All other changes, other than those that addressed other issues, are merely rebasing.

Reply:
ah, ya, I was going to point that out -- I wonder if we should disable the copy constructor in the dispatch stubs to make it clear doing this is wrong.

Reply:
If it is clearly wrong, I think we should (in a separate PR)

Reply:
sounds good.